src/common/meta/src/wal/meta.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snafu::OptionExt;
+
+use crate::error::{MissingKafkaTopicManagerSnafu, Result};
+use crate::kv_backend::KvBackendRef;
+use crate::wal::kafka::{KafkaTopic as Topic, KafkaTopicManager as TopicManager};
+use crate::wal::{WalOptions, WalProvider};
+
+/// The allocator responsible for allocating wal metadata for a table.
+#[derive(Default)]
+pub struct WalMetaAllocator {
```

Comment:
I prefer `Resource` instead of `Meta`. I read the context of the code, and it seems like something called  `RegionWalResouceAllocator`, but not the `WalMetaAllocator`.

Reply:
The `WalMetaAllocator` is now replaced with the `RegionWalOptionsAllocator`.